app/javascript/components/pages/DeckNew.js

```diff
@@ -8,32 +8,52 @@ export class DeckNew extends Component {
 			form: {
 				title: '',
 				description: '',
-				cards: [],
+				cards: [
+					{
+						term: '',
+						definition: '',
+					},
+					{
+						term: '',
+						definition: '',
+					},
+					{
+						term: '',
+						definition: '',
+					},
+				],
 			},
 			submitted: false,
 		}
 	}
 
 	handleTermChange = (e, cardNumber) => {
 		const { cards } = this.state.form
-		const { form } = this.state
+		let { form } = this.state
 		let currentCards = cards
 		let newCards = [...currentCards]
 		newCards[cardNumber] = { term: e.target.value, definition: '' }
-		let newForm = Object.assign({}, form)
-		newForm.cards = newCards
-		this.setState({ form: newForm })
+		form.cards = newCards
+		this.setState({ form: form })
 	}
 
-	handleDefChange = (e) => {
-		console.log('def', e.target.value)
+	handleDefChange = (e, cardNumber) => {
+		const { cards } = this.state.form
+		let { form } = this.state
+		let currentCards = cards
+		let newCards = [...currentCards]
+		const thisCard = cards[cardNumber]
+		newCards[cardNumber] = { ...thisCard, definition: e.target.value }
```

Comment:
awesome!

Reply:
since both the handleDefChange and handleTermChange mostly hold the same logic, this would be a good place to do refactoring so that you only need one function to do both. again: not an urgent change but something to think about.